dotnet/src/SemanticKernel.Core/Orchestration/FunctionExecutor.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.SemanticKernel.Orchestration;
+
+/// <summary>
+/// Kernel function executor implementation
+/// </summary>
+internal sealed class FunctionExecutor : IFunctionExecutor
+{
+    private readonly Kernel _originKernel;
+
+    internal FunctionExecutor(Kernel originKernel)
+    {
+        this._originKernel = originKernel;
+    }
+
+    /// <inheritdoc/>
+    public Task<KernelResult> ExecuteAsync(ISKFunction skFunction, ContextVariables variables, IReadOnlyFunctionCollection? functions = null, CancellationToken cancellationToken = default)
+    {
+        using var kernel = this._originKernel.Clone(functions);
```

Comment:
Do we have a case when we need to have a kernel with a different set of functions than the original one?

Reply:
Normally this can happen from the plan side, while you want to run from a specific set of functions, not all.